### ISSUE TITLE
Fundraiser for 2023

### ISSUE THREAD

**Issue by mj-xmr** [mj-xmr/SolOptXMR#147]
﻿Hey everybody and Merry Christmas,

This is a fundraiser for the further development of SolOptXMR, whose interface can be found below:

https://htmlpreview.github.io/?https://github.com/mj-xmr/wishlist-mj/master/html/index.html

# Intro

On the 11th Nov 2022, I published the release of the [Milestone 4 of 4](https://www.reddit.com/r/Monero/comments/ys49jx/soloptxmr_milestone_44_full_automation/) of the [SolOptXMR](https://github.com/mj-xmr/SolOptXMR) project.

As a side note, the last point of that report was "Business model introduction", which has been retraced during the Community Review, so it became irrelevant for the current state of SolOptXMR.

After completing my contracted parts, the current state of SolOptXMR could be described as:

- Stable: thanks to spending much more time than contracted on [unit & integration tests](https://github.com/mj-xmr/SolOptXMR/blob/master/docs/testing.md), which uncovered multiple edge cases tending to mutually destabilize the system whenever one of them got fixed. Now this is history and the tests run automatically at each commit in a short span of time, thus protecting the project from regressions in the future.
- Functional, yet having a development potential: most importantly the search algorithm's nature is purely random without any guidance. This classifies as a "naive search" (or "brute force"). As you might imagine, the search time for solutions in a wide search space as this one (many computers and long horizon), grows exponentially. We may retain the nature of the random search, which has great benefits in a multidimentional search space, yet provide it some guidance at the same time, since using the very same model, the algorithm may be replaced with either a [Monte Carlo simulation](https://github.com/mj-xmr/tsqsim/blob/master/src/lib-base/src/OptimizerBase.cpp#L114), that I already use in my other project, [tsqsim](https://github.com/mj-xmr/tsqsim), or using a [genetic algorithm](https://github.com/mj-xmr/galgo-2.0-mj) of a different Author, that I'm ready to integrate.

There are many more open points, implementation of which would greatly improve my software, that I hope is being used, or is at least planned to be used.

The full and regularly updated list can be found on the [project's Wiki](https://github.com/mj-xmr/SolOptXMR/wiki), just as a reference.

# Avoiding the CCS

I'd ask for the possibility of further development to be funded outside of the formerly used CCS ([Monero Community Crowdfunding System](https://ccs.getmonero.org/)), firstly due to its personal-related inefficiencies, that take up all my time and energy, that would otherwise be spent on the project, and secondly due to the fact, that the benefits that the CCS delivers, don't apply to me anymore, IMO. The main benefits are:

- The CCS serves as an escrow between the Donators and the Developers: I not only deliver everything that I promise on time, but I myself have already been acting as an escrow for another Community Member: u/jeonyiota, the [korean and greek translator](https://www.reddit.com/r/Monero/comments/z0dng5/manifesttims_translation_report_sepoctnov/).

If I wanted to rob anybody, this would have been my best chance, as Jiota continued her 4 month work only after the whole of the declared funds were collected on **my** account. I really just want to write my code and have nothing to do with the jealous Bad Actors, that currently plague the CCS (from my PoV) and don't "approve" other peoples' personal development, using political tactics as cheap as the infameous "cancel culture", "double speak" and simply by making rules as they go (in a nutshell).

- Community Reviews of the proposal itself: This normally protects the Donators from spending money on bogus proposals. My proposal has not only already been approved, but the Userbase expressed their positive attitude towards the project, which I'm very thankful for!

Here's a [perfectly fitting testimony](https://github.com/mj-xmr/SolOptXMR/issues/55#issuecomment-1155652582) of SolOptXMR itself, with all due respect to these proposals, which are Real:

> I didn't realize that it was funded thru CCS, as I've stopped following what goes on there since in most cases it seems like just a grabbing bazaar without much return, so I am rather happy that you seem to be an exception there to actually produce something of real value AND put monero first.

Observing the matter from a more quantitative PoV, here's the [cummulative Userbase's reaction](http://cryptog.hopto.org/monero/drama/reaction-milestone4-Screenshot_2022-12-22_18-59-40.png) after publishing the 4th Milestone. I guess, that the message is: "Shut up and keep writing that code".

In such fortunate case, the funding page can be [found here](https://htmlpreview.github.io/?https://github.com/mj-xmr/wishlist-mj/master/html/index.html).

The funding page contains each of the major points separated, as well as all the tiny tasks merged into larger ones. Naturally, the budget spared on a given task will be reused on others. The separation serves me as a way to prioritize my tasks according to your expectations.

In order to capture even more information than using the above method, for example either from my Users, who currently can't donate or optimistically - when all the points get fully funded, thus yielding no difference between their priorities, a mirror of the tasks is presented in this GitHub issue, where I'd ask you to vote on particular tasks individually, using the *Thumbs Up* feature's "Reactions", or even *Thumbs Down* if you're firmly against a particular feature.

ISSUE LINK

I will start working on any of the tasks, whichever collects the funding first. Specifically, I won't be waiting for the full budget to be accumulated.

# Endor's work

The planned work is absent of the tasks, which were explicitly reserved for endor, who firmly expressed the will of finalizing them until an undisclosed date. If you'd like to know more about the progress of those tasks alone, you have to ask endor directly. Endor's tasks are described in [the original proposal](https://ccs.getmonero.org/proposals/soloptxmr-mj-endor-2022.html).

For reference, the discussion, where endor made the public statement paraphrased by me above, can be traced [from here on](https://repo.getmonero.org/monero-project/ccs-proposals/-/merge_requests/299#note_19401). (NSFW warning!)

# Deadline

I'd like to take it easy and finalize all the declared tasks until the end of year 2023. The most critical features, that deal with helping you in protection of your mining hardware against damage and your electrical installation against catching fire across the warm time of the year, have already been implemented.

# Funds calculation

Assuming 270 working days in a year and an average of 2 hours per day of work on the project, as of 2022-12-26, I'd expect to collect a total sum of 225 XMR for the newly identified tasks.
